transcendence-app/src/auth/fortytwo/fortytwo.strategy.ts

```diff
@@ -0,0 +1,70 @@
+import { HttpService } from '@nestjs/axios';
+import { BadGatewayException, Injectable } from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+import { PassportStrategy } from '@nestjs/passport';
+import { plainToClass } from 'class-transformer';
+import { validate } from 'class-validator';
+import { Strategy } from 'passport-oauth2';
+import { lastValueFrom } from 'rxjs';
+import { CreateUserDto } from '../../users/dto/create-user.dto';
+import { UsersService } from '../../users/users.service';
+import { FortyTwoAuthConfig } from './fortytwo-config.interface';
+
+@Injectable()
+export class FortyTwoStrategy extends PassportStrategy(Strategy, '42') {
+  constructor(
+    private usersService: UsersService,
+    private httpService: HttpService,
+    protected configService: ConfigService<FortyTwoAuthConfig>,
+  ) {
+    super({
+      authorizationURL: 'https://api.intra.42.fr/oauth/authorize',
```

Comment:
I believe we should not hardcode urls or data prone to change, this should be in an env variable